src/api/route-services/pipelines.js

```diff
@@ -0,0 +1,196 @@
+    return {
+      Comment: 'N/A',
+      StartAt: 'DeleteCompletedJobs',
+      States: {
+        DeleteCompletedJobs: {
+          Type: 'Task',
+          Comment: 'Deletes all the preivous server jobs that are already completed.',
+          Resource: 'arn:aws:states:::eks:call',
+          Parameters: {
+            ClusterName: name,
+            CertificateAuthority: certAuthority,
+            Endpoint: endpoint,
+            Method: 'DELETE',
+            Path: `/apis/batch/v1/namespaces/${namespace}/jobs`,
+            QueryParameters: {
+              fieldSelector: [
+                'status.successful=1',
+              ],
+            },
+          },
+          Next: 'LaunchKubernetesJobIfNotExists',
+        },
+        LaunchKubernetesJobIfNotExists: {
+          Type: 'Task',
+          Comment: 'Attempts to create a Kubernetes Job for the pipeline server. Will swallow a 409 (already exists) error.',
+          Resource: 'arn:aws:states:::eks:call',
+          Parameters: {
+            ClusterName: name,
+            CertificateAuthority: certAuthority,
+            Endpoint: endpoint,
+            Method: 'POST',
+            Path: `/apis/batch/v1/namespaces/${namespace}/jobs`,
+            RequestBody: {
+              apiVersion: 'batch/v1',
+              kind: 'Job',
+              metadata: {
+                name: `remoter-server-${experimentId}`,
+              },
+              spec: {
+                template: {
+                  metadata: {
+                    name: `remoter-server-${experimentId}`,
+                  },
+                  spec: {
+                    containers: [
+                      {
+                        name: 'remoter-server',
+                        image,
+                      },
+                    ],
+                    restartPolicy: 'Never',
+                  },
+                },
+              },
+            },
+          },
+          Retry: [
+            {
+              ErrorEquals: ['EKS.409'],
+              IntervalSeconds: 1,
+              BackoffRate: 2.0,
+              MaxAttempts: 2,
+            },
+          ],
+          Catch: [
+            {
+              ErrorEquals: ['EKS.409'],
+              ResultPath: '$.error-info',
+              Next: 'Wait',
+            },
+          ],
+          Next: 'Wait',
+        },
+        Wait: {
+          Type: 'Wait',
+          Seconds: 5,
+          End: true,
+        },
+      },
+    };
+  }
+
+
+  async create(experimentId) {
+    const {
+      roleArn, accountId,
+    } = await this.getClusterInformation();
+
+    const { clusterEnv, sandboxId } = config;
+
+    const pipelineHash = crypto
+      .createHash('sha1')
+      .update(`${experimentId}-${sandboxId}`)
+      .digest('hex');
+
+    const params = {
+      name: `biomage-pipeline-${pipelineHash}`,
+      roleArn,
+      definition: JSON.stringify(await this.generatePipeline(experimentId)),
+      loggingConfiguration: {
+        level: 'OFF',
```

Comment:
Logging to where is this for? Why is it off?

Reply:
This is for logging to CloudTrail. I don't think we need this for now, as the actual results from the steps, as well as their inputs, still get persisted in the execution.

Reply:
yup, we don't need it for now.